extensions/amp-video/0.1/flexible-bitrate.js

```diff
@@ -234,11 +240,17 @@ export class BitrateManager {
         this.videos_.splice(i, 1);
         return;
       }
-      if (!video.paused) {
+      if (
+        !video.paused ||
+        getBufferedPercentage(video) > BUFFERED_THRESHOLD_PERCENTAGE
+      ) {
         return;
       }
       this.sortSources_(video);
-      video.load();
+      // If video has metadata, load again. Avoid loading videos that are far or were unloaded.
```

Comment:
We'll have to spend time testing this but from what I know, not calling `load()` will not restart the media selection process, and so even if you sort the sources, it won't affect which source plays on `play()`.

tldr: you need to call load() at some point, if it becomes too tricky let's save this optimization for a follow up PR with dedicated testing

Reply:
Sounds good. I think this optimization would only be useful if the mediapool replaces a video but doesn't call load (so it will call it later), but it shouldn't get triggered with today's code anyways